src/kernel/scheduling/scheduler.c

```diff
@@ -139,6 +139,14 @@ cpu_status_t* schedule(cpu_status_t* cur_status) {
    return cur_status;
 }*/
 
+void scheduler_add_task(task_t* task) {
+    task->next = root_task;
+    root_task = task;
+    if (root_task == NULL) {
```

Comment:
this will never happen, since you assign `root_task` before the if. I would suggest moving line 144 after the if :)